In [89]:
import pandas as pd #to load json data to dataframe and iterate,manipulate through json data

In [90]:
#loads data to dataframe -> SearchData and displays first 100 of them
SearchData = pd.read_json('searches.json', lines=True)
SearchData.head(100)

is_instructor  login_count  search_count       uid
0            True            1             2   6061521
1           False            4             0  11986457
2           False            1             0  15995765
3            True            3             0   9106912
4           False            1             0   9882383
5           False            1             0   5333200
6           False            2             0   3583107
7            True            1             0  11760157
8           False            3             0  13150423
9           False            4             2   8004609
10          False            1             3  13536345
11          False            2             4     52868
12          False            6             0   9613243
13           True           14             0  12986377
14           True            1             0   9792541
15          False            4             0   2815244
16          False            7             0  13320527
17          False            1             0   5651994
18          False            1             0  12260537
19           True            7             0  14915463
20           True            7             1   8228927
21           True            1             0   3378724
22          False            1             0   5936973
23           True            4             0   8373533
24          False            1             0  16102156
25          False            4             2   9185533
26           True            4             0   8905420
27          False            1             1   9954854
28          False            6             0  14133079
29          False            1             0   4716739
..            ...          ...           ...       ...
70          False            1             0  15299641
71           True            1             0  16691979
72          False            3             0   6026021
73          False            1             0   5573213
74          False            2             0   7248535
75          False            3             0  14341169
76           True            3             2    268430
77          False            2             0   1082427
78           True            4             0  11360517
79           True            2             1   3919480
80          False            1             0   8186038
81           True            1             0   4691263
82           True            1             0   7355273
83          False            2             1  13945044
84          False            2             0   8648947
85           True            2             2  14955604
86           True            3             1  10372718
87          False            2             0  13124596
88          False            1             0  13170480
89           True            1             2  10530538
90          False            1             3   8024978
91           True            3             0  15491559
92          False            1             0  13825496
93          False            2             0   4197889
94          False            5             0   9722213
95          False            6             0  14629029
96          False            1             0  13485368
97          False            2             0   9656655
98          False            5             0  14759972
99           True            1             1  11164560

[100 rows x 4 columns]

In [96]:
#creating a dictionary of users who were shown the design B 
#containing user id (uid) as key & 
#number of times the user used the search feature (search_count)as value
b_user = {}
for index, row in SearchData.iterrows():
    b = row['uid']
    if(b%2 != 0):
        b_user[b] = row['search_count']
b_user
    

{39043: 0,
 174623: 0,
 175145: 0,
 203495: 0,
 252639: 0,
 325327: 0,
 351701: 0,
 368693: 0,
 429071: 0,
 509223: 0,
 547507: 0,
 596881: 0,
 608185: 0,
 659849: 0,
 674805: 5,
 699035: 0,
 800163: 0,
 853849: 0,
 917351: 0,
 955085: 0,
 1045725: 0,
 1082427: 0,
 1112901: 3,
 1141181: 0,
 1177099: 0,
 1188165: 0,
 1202645: 0,
 1203395: 0,
 1215461: 0,
 1400519: 0,
 1440233: 0,
 1475209: 0,
 1478415: 0,
 1547629: 0,
 1572895: 2,
 1585489: 2,
 1638057: 0,
 1638829: 0,
 1657315: 2,
 1668679: 0,
 1724865: 0,
 1791653: 3,
 2013311: 0,
 2036945: 0,
 2043101: 0,
 2180213: 0,
 2183257: 0,
 2321009: 0,
 2327303: 0,
 2368013: 9,
 2615111: 1,
 2730735: 0,
 2849545: 0,
 2898229: 3,
 3013413: 0,
 3372619: 0,
 3381743: 0,
 3400195: 0,
 3402261: 0,
 3469933: 4,
 3494291: 2,
 3522037: 0,
 3544903: 0,
 3546309: 1,
 3583107: 0,
 3596125: 1,
 3691369: 5,
 3701785: 3,
 3754509: 0,
 3761361: 1,
 3793213: 0,
 3857555: 0,
 3942521: 0,
 4039745: 0,
 4068379: 6,
 4120397: 0,
 4197889: 0,
 4310571: 1,
 433360

In [92]:
#creating a dictionary of users who were shown the design A 
#containing user id (uid) as key & 
#number of times the user used the search feature (search_count)as value
a_user = {}
for index, row in SearchData.iterrows():
    a = row['uid']
    if(a%2 == 0):
        a_user[a] = row['search_count']
a_user

{8134: 2,
 52868: 4,
 87006: 0,
 174594: 8,
 193986: 0,
 204924: 0,
 268430: 2,
 270944: 3,
 311808: 0,
 325848: 0,
 387824: 2,
 458564: 0,
 495192: 0,
 596434: 0,
 866028: 0,
 1032514: 2,
 1037096: 3,
 1079304: 0,
 1108670: 9,
 1117598: 2,
 1184250: 0,
 1205992: 2,
 1218512: 0,
 1240360: 0,
 1381208: 0,
 1401104: 3,
 1622454: 2,
 1706436: 0,
 1926246: 3,
 1958906: 2,
 1962364: 0,
 2028090: 0,
 2042472: 0,
 2051872: 0,
 2052840: 8,
 2102926: 0,
 2122824: 0,
 2128420: 0,
 2129136: 0,
 2142572: 0,
 2247594: 0,
 2272480: 1,
 2465070: 8,
 2760652: 5,
 2776458: 0,
 2781844: 0,
 2815244: 0,
 2861216: 1,
 2903266: 4,
 2990790: 0,
 3008340: 0,
 3012868: 0,
 3128388: 0,
 3142498: 0,
 3170318: 0,
 3180160: 8,
 3258202: 4,
 3302590: 0,
 3305956: 0,
 3365272: 1,
 3378724: 0,
 3476168: 0,
 3562222: 0,
 3637508: 0,
 3644270: 0,
 3686072: 1,
 3725688: 0,
 3751590: 1,
 3815382: 0,
 3816152: 0,
 3860260: 0,
 3878166: 0,
 3919480: 1,
 3996560: 0,
 4008090: 5,
 4034102: 0,
 4179688: 0,
 4197454: 0,
 4206

In [93]:
#Creating a cumulative count of the total number of times search bar was used by
#users viewing design B (b_search_count) & 
#number of times search bar was used by users viewing design A (a_search_count)
b_search_count = sum(b_user.values())
a_search_count = sum(a_user.values())
print("A search count",a_search_count)
print("B search count",b_search_count)

A search count 266
B search count 231


In [94]:
#Finding the frequency of use of search bar in design B vs 
#frequency of use of search bar in design A
a_size = len(a_user)
b_size = len(b_user)
print(a_size)
print(b_size)
a_search_frequency = a_search_count/a_size
b_search_frequency = b_search_count/b_size
print("A search frequency",a_search_frequency)
print("B search frequency",b_search_frequency)

333
348
A search frequency 0.7987987987987988
B search frequency 0.6637931034482759


Inference :-
1. Users using design A have used search bar more number of times over users using design B
(266 vs 231)
2. Frequency of use of search bar in design A is higher than Frequency of use of search bar in design B
(0.79 vs 0.66) 
Results :-
1. Did more users use the search feature in the new design (B)? 
ANS : NO
2. Did users search more often in the new design (B)?
ANS : NO